src/tools/compiletest/src/runtest.rs

```diff
@@ -3124,15 +3124,17 @@ impl<'test> TestCx<'test> {
     fn check_mir_dump(&self) {
         let test_file_contents = fs::read_to_string(&self.testpaths.file).unwrap();
 
-        let mut test_dir = self.testpaths.file.with_extension("");
+        let test_dir = self.testpaths.file.parent().unwrap();
 
+        let mut bit_width = "all";
         if test_file_contents.lines().any(|l| l == "// EMIT_MIR_FOR_EACH_BIT_WIDTH") {
-            test_dir.push(get_pointer_width(&self.config.target))
+            bit_width = get_pointer_width(&self.config.target);
         }
 
         if self.config.bless {
-            let _ = std::fs::remove_dir_all(&test_dir);
+            let _ = std::fs::remove_dir_all(&self.testpaths.file.with_extension(""));
```

Comment:
huh, this does anything? don't we need to glob all such files?

Reply:
that needs to be removed. it was to clean up the old files during the transition.

Reply:
The problem is that without any removal, we'll end up with orphaned files (e.g. when the `// EMIT_MIR` command is changed, the old files would stay). But with the new scheme, we can't just remove all `filename.*` where `*` is not `rs`, because 32 bit and 64 bit generation happens in different runs, so we'd keep removing the files from the other bit width.

I'm not sure what to do here, except maybe ignore all files that don't have the bitwidth that is currently being tested.

Reply:
it's already the case that we had orphaned files so the previous code wasn't bullet proof. With the current code can't we just remove `expected_file`?

Reply:
ah no I see, you mean if we change the name of the tested function we won't clean up the old expectations. hmmm.

Reply:
we could nuke everything with the same crate name? it's an over approximation but not too bad?

Reply:
same problem as removing the crate name in `EMIT_MIR` how do we know which files belong to which tests? Especially when they don't necessarily start with that test's crate prefix, ie: `src/test/mir-opt/core.ops-function-Fn-call.AddMovesForPackedDrops.before.mir.all`

Reply:
O_o I did not think of that. I guess we need to prefix the name to everything that doesn't already have it prefixed?

Reply:
yea i guess so

Reply:
though there is the problem of knowing _what_ to prefix :/ again, the files start with crate names which are different from the source file names.

Reply:
ok I've used the `glob` crate to remove all files starting with the crate prefix of the test. The next step would be to prefix expectations that are from other crates.

Reply:
done! i made sure that all files are prefixed with the cratename.